### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by Log-benjamin**
## [Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#project-requirements)

- Create a new Rails app called 'hello-rails-react'.
   - Use the command rails new -j web pack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the bundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel-loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.

### CODE REVIEW

File: app/javascript/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import store from '../store/store';
+import { Provider } from 'react-redux';
+import Greeting from './Greeting.jsx';
+
+const App = () => {
+  return(
+     <Provider store={store}>
+       <Router>
+         <Routes>
+            <Route path='/' element={<Greeting />}></Route>
```
Comment (DammyShittu) [inline app/javascript/components/App.jsx:12]: - [ ] When I load up your app and visit the index page, nothing is displayed. When I inspect the server logs, there is no API call being made either. Kindly ensure the API is being called and a random message is displayed on the browser.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Log-benjamin,

## Highlights
✔️ Webpack feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Log-benjamin** [comment]
- To see the json file from the API
http://localhost:3000/api/v1/index
Screenshot (141)

Screenshot (144)

**Review by topeogunleye — APPROVED** [review]
### Hi @Log-benjamin, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 

- **Swift Issue Resolution**: Great job addressing the feedback from the previous review. Addressing such feedback in a timely manner ensures a smooth and efficient development process.
- **API Integration**: Kudos for successfully ensuring the API call is made on the index page. Seamless API integration is crucial for the dynamic behavior of many modern apps, and it's wonderful to see it working in your application.
- **Display Enhancements**: Well done on making sure the random message is displayed in the browser. Having visible feedback for user actions, like API call results, significantly enhances user experience.
- **Attention to Detail**: Your commitment to diving deep into the issue, understanding the root cause, and rectifying it is commendable. It showcases a strong dedication to code quality and functionality.

Keep pushing forward! 🥳 Your dedication to refining the project and addressing critical issues will certainly set you up for success in future endeavors. 🚀
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Log-benjamin** [comment]
@topeogunleye, Thank you for taking the time to review my project.

**Comment by topeogunleye** [comment]
> @topeogunleye, Thank you for taking the time to review my project.

@Log-benjamin, you're welcome

**Pull request merged**